api/src/main/java/org/pmiops/workbench/api/ProfileController.java

```diff
@@ -625,6 +630,8 @@ private boolean userNeverLoggedIn(
     validateProfileFields(updatedProfile);
     DbUser user = userProvider.get();
 
+    // save previous profile for audit trail
+    final Profile previousProfile = profileService.getProfile(user);
     if (!userProvider.get().getGivenName().equalsIgnoreCase(updatedProfile.getGivenName())
```

Comment:
If we're saving the previous profile, we may be able to replace some of these userProvider accesses with previous profile accesses (which semantically make more sense)